code/Test_definitions/connected-network-type-subscriptions.feature

```diff
@@ -1,289 +1,488 @@
-  Scenario: Create subscription with invalid parameter
-    Given the request body is not compliant with the schema "/components/schemas/SubscriptionRequest"
+  @connected_network_type_subscriptions_C01.01_device_empty
+  Scenario: The device value is an empty object
+    Given the header "Authorization" is set to a valid access token which does not identify a single device
+    And the request body property "$.device" is set to: {}
     When the request "createConnectedNetworkTypeSubscription" is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_C01.02_device_identifiers_not_schema_compliant
+  Scenario Outline: Some device identifier value does not comply with the schema
+    Given the header "Authorization" is set to a valid access token which does not identify a single device
+    And the request body property "<device_identifier>" does not comply with the OAS schema at "<oas_spec_schema>"
+    When the request "createConnectedNetworkTypeSubscription" is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | device_identifier          | oas_spec_schema                             |
+      | $.device.phoneNumber       | /components/schemas/PhoneNumber             |
+      | $.device.ipv4Address       | /components/schemas/DeviceIpv4Addr          |
+      | $.device.ipv6Address       | /components/schemas/DeviceIpv6Address       |
+      | $.device.networkIdentifier | /components/schemas/NetworkAccessIdentifier |
+
+  # This scenario may happen e.g. with 2-legged access tokens, which do not identify a single device.
+  @connected_network_type_subscriptions_C01.03_device_not_found
+  Scenario: Some identifier cannot be matched to a device
+    Given the header "Authorization" is set to a valid access token which does not identify a single device
+    And the request body property "$.device" is compliant with the schema but does not identify a valid device
+    When the request "createConnectedNetworkTypeSubscription" is sent
+    Then the response status code is 404
+    And the response property "$.status" is 404
+    And the response property "$.code" is "IDENTIFIER_NOT_FOUND"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_C01.04_unnecessary_device
+  Scenario: Device not to be included when it can be deduced from the access token
+    Given the header "Authorization" is set to a valid access token identifying a device
+    And the request body property "$.device" is set to a valid device
```

Comment:
Clarification that the test applies whether the same device is identified twice, or two different devices are identified
```suggestion
    And the request body property "$.device" is also set to a valid device, which may or may not be the same device
```